src/Image.php

```diff
@@ -381,6 +410,55 @@ public function quality($quality)
     }
 
     /**
+     * Add a watermark to current image
+     *
+     * @param string|object $watermark Image to set as watermark
+     * @param array         $settings  Overwrite default settings
+     *
+     * @return self
+     */
+    public function watermark($watermark, $settings = [])
+    {
+        if (is_string($watermark)) {
+            $watermark = self::fromFile($watermark);
+        }
+
+        $settings = array_merge(self::$watermarkSettings, $settings);
```

Comment:
Not sure whether array_merge is the best choice in this case. `array_replace` seems better, or, simply:
```php
$settings += self::$watermarkSettings;
```